### PR TITLE
HTTPS support - via Ingress

### PR DESCRIPTION
**Pull request by carlos-logro**
issue: #370 

# Changes
Made a script that automates the creation of a TLS certificate & applies it using ingress.
The script needs to be modified by the user with the relevant information to create the certificate.
Its worth mentioning that the key & certificate created are stored under the `etc/ssl/certs` directory. 

# Submitter Checklist

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

### CODE REVIEW

**Comment by carlos-logro** [comment]
@AlanGreene @a-roberts @mnuttall

File: ingress/ingress-https-setup.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+# path to cert & keys
+export CERTIFICATE_DIR="/etc/ssl/certs/tekton-dashboard"
+
+# certificate data
+# *Make sure names contain only lowercase alphanumeric characters, . or -. Must start & end with alphanumeric characters*
+export CERTIFICATE_KEY=""
+export CERTIFICATE_KEY_PASSPHRASE=""
+export CERTIFICATE_NAME=""
+export CERTIFICATE_SECRET_NAME=""
+export IP_ADDRESS=$(ip route get 8.8.8.8 | awk '{print $NF; exit}')
```
Comment (akihikokuroda) [inline ingress/ingress-https-setup.sh]: The IP address must be one of host within the kubernetes cluster.  Should this be executed one of the cluster hosts?

File: ingress/ingress-https-setup.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# path to dashboard repo
+export REPO_TOP_LEVEL=$(git rev-parse --show-toplevel)
+# path to cert & keys
+export CERTIFICATE_DIR="/etc/ssl/certs/tekton-dashboard"
```
Comment (akihikokuroda) [inline ingress/ingress-https-setup.sh]: I guess that this directory can be accessible by the admin user.  Should this be executed by this admin user?
The certificate can be generated in any directory.

File: README.md [hunk 3]
```diff
@@ -109,6 +109,17 @@ privileged         true      [*]       RunAsAny    RunAsAny           RunAsAny
 restricted         false     []        MustRunAs   MustRunAsRange     MustRunAs   RunAsAny    <none>     false            [configMap downwardAPI emptyDir persistentVolumeClaim projected secret]
 ```
 
+### Use HTTPS on OpenShift via Ingress
+#### Pre-requisites:
```
Comment (akihikokuroda) [inline README.md]: I believe that "Enable TLS for dashboard access via Ingress" is better title for this.

File: ingress/ingress-https-setup.sh (hunk 2 above)
Comment (carlos-logro) [reply ingress/ingress-https-setup.sh]: What do you mean by "executed by this admin user"?

File: ingress/ingress-https-setup.sh (hunk 1 above)
Comment (carlos-logro) [reply ingress/ingress-https-setup.sh]: I believe it should be executed in the cluster the hosts wants to access in the browser

Comment (akihikokuroda) [reply ingress/ingress-https-setup.sh]: OK.  Would you put the assumption in the README.md?

File: ingress/ingress-https-setup.sh (hunk 2 above)
Comment (akihikokuroda) [reply ingress/ingress-https-setup.sh]: I should have put "root" user.  I'm not sure if the regular user can create a directory in the /etc directory.

**Pull request merged**